src/GraphQL.Client/GraphQLHttpClientOptions.cs

```diff
@@ -61,9 +61,17 @@ public class GraphQLHttpClientOptions
     /// Note that compatible to the draft graphql-over-http spec GraphQL Server MAY return 4xx status codes (401/403, etc.)
     /// with well-formed GraphQL response containing errors collection.
     /// </summary>
-    public Func<HttpResponseMessage, bool> IsValidResponseToDeserialize { get; set; } = r =>
-        // Why not application/json? See https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#processing-the-response
-        r.IsSuccessStatusCode || r.StatusCode == HttpStatusCode.BadRequest || r.Content.Headers.ContentType?.MediaType == "application/graphql+json";
+    public Func<HttpResponseMessage, bool> IsValidResponseToDeserialize { get; set; } = DefaultIsValidResponseToDeserialize;
+
+    private static IReadOnlyCollection<string> AcceptedResponseContentTypes { get; } = new[] { "application/graphql+json", "application/json", "application/graphql-response+json" };
```

Comment:
```suggestion
    private static readonly IReadOnlyCollection<string> AcceptedResponseContentTypes = new[] { "application/graphql+json", "application/json", "application/graphql-response+json" };
```

Lets reuse that collection... (please adapt the filed name to the configured rules after changing that)

Reply:
yeah, good point, it was stuck in between of being a field and being actually configurable :P I'd make it a non static field of options possibly, but then we'd need to rewrite anther part of this class, not worth most likely